Examples/Io/Root/include/ActsExamples/Io/Root/RootDigitizationWriter.hpp

```diff
@@ -0,0 +1,238 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2017-2018 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "ActsExamples/Digitization/SmearingAlgorithm.hpp"
+#include "ActsExamples/EventData/DigitizedHit.hpp"
+#include "ActsExamples/Framework/WriterT.hpp"
+#include <Acts/EventData/Measurement.hpp>
+#include <Acts/Geometry/GeometryContext.hpp>
+#include <Acts/Geometry/GeometryHierarchyMap.hpp>
+#include <Acts/Geometry/GeometryIdentifier.hpp>
+#include <Acts/Utilities/Helpers.hpp>
+#include <Acts/Utilities/ParameterDefinitions.hpp>
+
+#include <TTree.h>
+
+#include <memory>
+#include <mutex>
+
+class TFile;
+class TTree;
+
+namespace ActsExamples {
+
+/// @class RootDigitizationWriter
+///
+/// Write out a planar cluster collection into a root file
+/// to avoid immense long vectors, each cluster is one entry
+/// in the root file for optimised data writing speed
+/// The event number is part of the written data.
+///
+/// A common file can be provided for to the writer to attach his TTree,
+/// this is done by setting the Config::rootFile pointer to an existing file
+///
+/// Safe to use from multiple writer threads - uses a std::mutex lock.
+class RootDigitizationWriter
+    : public WriterT<GeometryIdMultimap<
+          Acts::FittableMeasurement<ActsExamples::DigitizedHit>>> {
+ public:
+  struct Config {
+    /// Which measurement collection to write.
+    std::string inputMeasurements;
+    /// Which simulated (truth) hits collection to use.
+    std::string inputSimulatedHits;
+    std::string filePath = "";          ///< path of the output file
+    std::string fileMode = "RECREATE";  ///< file access mode
+
+    TFile* rootFile = nullptr;  ///< common root file
+    /// Optional the smearFunctions
+    Acts::GeometryHierarchyMap<SmearingAlgorithm::SupportedSmearer> smearers;
+  };
+
+  struct DigitizationTree {
+    const std::array<std::string, Acts::eBoundSize> bNames = {
+        "l0", "l1", "phi", "theta", "qop", "time"};
```

Comment:
```suggestion
        "loc0", "loc1", "phi", "theta", "qoverp", "time"};
```
Suggestion for consistency w/ the internal naming/enum values.

Reply:
makes sense.